namesys/publisher.go

```diff
@@ -295,12 +303,30 @@ func selectRecord(recs []*pb.IpnsEntry, vals [][]byte) (int, error) {
 
 // ValidateIpnsRecord implements ValidatorFunc and verifies that the
 // given 'val' is an IpnsEntry and that that entry is valid.
-func ValidateIpnsRecord(k string, val []byte) error {
+func ValidateIpnsRecord(r *record.ValidationRecord) error {
+	if r.Namespace != "ipns" {
+		return ErrInvalidPath
+	}
+
 	entry := new(pb.IpnsEntry)
-	err := proto.Unmarshal(val, entry)
+	err := proto.Unmarshal(r.Value, entry)
 	if err != nil {
 		return err
 	}
+
+	// Note: The DHT will actually check the signature so we don't
+	// need to do that here
+
+	// Author in key must match author in record
+	pid, err := peer.IDFromString(r.Key)
+	if err != nil {
+		return ErrInvalidAuthor
+	}
+	if string(pid) != string(r.Author) {
```

Comment:
Do we actually need to call `string` here?

Reply:
No, we should be able to get rid of that cast once the go-libp2p-peer lib is updated

Reply:
Got it.